airflow/api/common/experimental/trigger_dag.py

```diff
@@ -25,14 +25,19 @@
 from airflow.utils.state import State
 
 
-def trigger_dag(dag_id, run_id=None, conf=None, execution_date=None,
-                replace_microseconds=True):
-    dagbag = DagBag()
-
-    if dag_id not in dagbag.dags:
+def _trigger_dag(
```

Comment:
I don't see a need of new function here. Would it not be possible to add this code to `trigger_dag` itself?

Reply:
Well it makes it easier for testing as we can inject mocked `dag_run`, `dag_bag`, and 'dag'.